### PR TITLE
Inventory UI

### PR DESCRIPTION
**Pull request by RezaKusha**

### CODE REVIEW

File: Assets/Animations/Inventory/Hide-Inventory.anim
Comment (URAmiRBin) [inline Assets/Animations/Inventory/Hide-Inventory.anim:1]: please use dotween to do animations like this since they are more optimized and programmers have more control on it so we don't have to edit it every time we make a change.
https://dotween.demigiant.com/

File: Assets/GUI Elements/UI collection Red&Gold/Frame_big.png.meta
Comment (URAmiRBin) [inline Assets/GUI Elements/UI collection Red&Gold/Frame_big.png.meta:1]: i dont see any reason these meta files are commited in this branch, make another branch and make a separate pull request if necessary

File: Assets/Scripts/Interaction System/Interactable.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class Interactable : MonoBehaviour
         public UnityEvent<Character> onInteraction;
 
         private bool _canBeInteracted;
-        private Character _interactor;
+        public Character _interactor;
```
Comment (URAmiRBin) [inline Assets/Scripts/Interaction System/Interactable.cs]: This only depends on the interactable itself. no need to make it public

File: Assets/Scripts/Inventory/Character.cs [hunk 2]
```diff
@@ -4,7 +4,14 @@ public class Character : MonoBehaviour, InventorySystemCharacter
...
 
-    public Inventory Inventory => _inventory;
+    public Inventory Inventory
+    {
+        get
+        {
+            var inventory = _inventory;
+            return inventory;
+        }
+    }
```
Comment (URAmiRBin) [inline Assets/Scripts/Inventory/Character.cs]:
```suggestion
    public Inventory Inventory => _inventory;
```

its the same thing, dont make to too complicated

File: Assets/Scripts/Inventory/Inventory.cs [hunk 3]
```diff
@@ -24,5 +30,7 @@ public void RemoveItem(ItemData itemData)
             else
                 items.Remove(itemName);
         }
+        
+        //OnInventoryChange.Invoke(items);
```
Comment (URAmiRBin) [inline Assets/Scripts/Inventory/Inventory.cs]: why is this commented out?

File: Assets/Scripts/InventoryAnimation.cs
Comment (URAmiRBin) [inline Assets/Scripts/InventoryAnimation.cs:1]: as i said, please use dotween

File: Assets/TextMesh Pro/Shaders/TMP_Bitmap-Custom-Atlas.shader
Comment (URAmiRBin) [inline Assets/TextMesh Pro/Shaders/TMP_Bitmap-Custom-Atlas.shader:1]: these should be added in another pr

File: Assets/Sprites/Map.png
Comment (URAmiRBin) [inline Assets/Sprites/Map.png:1]: this should be in the same folder as other items, not here

File: Assets/Scripts/ItemPresenter.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            // amountText.text = amount.ToString();
+        }
+
+        public void Clear()
+        {
+            itemName = "";
+            itemImage = null;
+            
+            // TODO: Reset the image
+            // TODO: Reset the counter text
```
Comment (URAmiRBin) [inline Assets/Scripts/ItemPresenter.cs]: remove the comments

File: Assets/Scripts/ItemPresenter.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+        private string itemName;
+
+        public string ItemName => itemName;
+
+        public void Setup(string itemName, Sprite image/*, int amount = 1*/)
+        {
+            this.itemName = itemName;
+            itemImage.sprite = image;
+            // amountText.text = amount.ToString();
```
Comment (URAmiRBin) [inline Assets/Scripts/ItemPresenter.cs]: remove the comments

File: Assets/Scripts/ItemPresenter.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    public class ItemPresenter : MonoBehaviour
+    {
+        [SerializeField] private Image itemImage;
+        //[SerializeField] private Text amountText;
+
+        private string itemName;
+
+        public string ItemName => itemName;
+
+        public void Setup(string itemName, Sprite image/*, int amount = 1*/)
```
Comment (URAmiRBin) [inline Assets/Scripts/ItemPresenter.cs]: add a todo instead of commenting

File: Assets/Scripts/InventoryUI.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            // slot.GetComponentInChildren<Image>().sprite = null;
+            slot.Clear();
+        }
+    }
+
+    private void SelectItemAtIndex(int index)
+    {
+        SelectedItem = slots[index].ItemName;
+        Debug.Log("picked");
```
Comment (URAmiRBin) [inline Assets/Scripts/InventoryUI.cs]: remove debug logs, they are only for your own tests

File: Assets/Scripts/InventoryUI.cs [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            // TODO: Set the count UI to item.Value
+            i++;
+        }
+    }
+
+    private void ClearAllSlots()
+    {
+        foreach (var slot in slots)
+        {
+            // slot.GetComponentInChildren<Image>().sprite = null;
```
Comment (URAmiRBin) [inline Assets/Scripts/InventoryUI.cs]: remove comments

File: Assets/Scripts/InventoryUI.cs [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            var slot = slots[i];
+            var t = i;
+            slot.GetComponentInChildren<Button>().onClick.AddListener(() => SelectItemAtIndex(t));
+        }
+    }
+
+    public void UpdateSlots(Dictionary<string, int> items)
+    {
+        //ClearAllSlots();
```
Comment (URAmiRBin) [inline Assets/Scripts/InventoryUI.cs]: why don't you clear?

**Review by URAmiRBin — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Inventory/Character.cs (hunk 2 above)
Comment (RezaKusha) [reply Assets/Scripts/Inventory/Character.cs]: done

File: Assets/Scripts/Interaction System/Interactable.cs (hunk 1 above)
Comment (RezaKusha) [reply Assets/Scripts/Interaction System/Interactable.cs]: done

File: Assets/Scripts/Inventory/Inventory.cs (hunk 3 above)
Comment (RezaKusha) [reply Assets/Scripts/Inventory/Inventory.cs]: fixed

**Review by URAmiRBin — APPROVED** [review]
